src/k2/dto/FieldTypes.h

```diff
@@ -34,11 +34,15 @@ namespace dto {
 
 enum class FieldType : uint8_t {
     NULL_T = 0,
-    STRING = 1, // NULL characters in string is OK
-    INT16T = 2, 
-    INT32T = 3, 
-    INT64T = 4,
-    FLOAT = 5, // Not supported as key field for now
+    STRING, // NULL characters in string is OK
+    INT16T,
+    INT32T,
+    INT64T,
+    FLOAT, // Not supported as key field for now
+    DOUBLE,  // Not supported as key field for now
+    BOOL,
+    FIELD_TYPE,
```

Comment:
What is this one for?

Reply:
This is needed for determining the type of a reference field as part of an expression. We discussed this being the way users can perform schema migrations by discovering records where the desired column isn't upgraded

Reply:
I still don't understand. I don't see it used anywhere unless I missed it. We can talk about it next week.